generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -75,14 +79,47 @@ void WriteCSV(Cont const & cont, string const & fileName)
     fout << a.first << "," << a.second << endl;
 }
 
-/// \returns expected point altitude in meters according to linear model.
-double ExpectedPointAltitude(int16_t startAltitudeMeters, int16_t endAltitudeMeters,
-                             double distFromStartMeters, double featureLengthMeters)
+/// \returns y = k * x + b. It's an expected altitude in meters.
+double ExpectedPointAltitude(double k, double b, double distFromStartMeters)
 {
-  if (featureLengthMeters == 0.0)
-    return 0;
-  double const k = (endAltitudeMeters - startAltitudeMeters) / featureLengthMeters;
-  return startAltitudeMeters + k * distFromStartMeters;
+  return k * distFromStartMeters + b;
+}
+
+template <class T>
+long double SumWithDivisor(vector<T> const & vec, long double divisor)
+{
+  long double result = 0.0;
+  for (auto item : vec)
+    result += item/divisor;
+  return result;
+}
+
+/// \brief Calculates factors |k| and |b| of a line using linear least squares method.
+/// \returns false in case of error (e.g. if the line is parallel with ordinate axis)
+/// and true otherwise.
+bool LinearLeastSquaresFactors(vector<double> const & xs, vector<double> const & ys, double & k, double & b)
+{
```

Comment:
Much easier, right? Coincides with https://en.wikipedia.org/wiki/Simple_linear_regression and most statistics literature.